daemon/k8s_watcher.go

```diff
@@ -932,11 +937,61 @@ func (d *Daemon) addCiliumNetworkPolicy(obj interface{}) {
 		}
 	}
 
+	var cnpns k8s.CiliumNetworkPolicyNodeStatus
 	if err != nil {
-		log.Warningf("Unable to add CiliumNetworkPolicy %s: %s", rule.Metadata.Name, err)
+		cnpns = k8s.CiliumNetworkPolicyNodeStatus{
+			OK:          false,
+			Error:       fmt.Sprintf("%s", err),
+			LastUpdated: time.Now(),
+		}
+		log.Warningf("Unable to add CiliumNetworkPolicy %s: err: '%s'. err != nil: '%t'. a nil object: '%s'", rule.Metadata.Name, err, err != nil, nil)
 	} else {
+		cnpns = k8s.CiliumNetworkPolicyNodeStatus{
+			OK:          true,
+			Error:       "OK",
+			LastUpdated: time.Now(),
+		}
 		log.Infof("Imported CiliumNetworkPolicy %s", rule.Metadata.Name)
 	}
+
+	rule.SetPolicyStatus(nodeaddress.GetName(), cnpns)
+	_, err = crdClient.Update(rule)
+	if err != nil {
+		ns := k8s.ExtractNamespace(&rule.Metadata)
+		name := rule.Metadata.GetObjectMeta().GetName()
+		log.Warningf("k8s: unable to update CNP %s/%s with status: %s, retrying...", ns, name, err)
+		go func(ns, name string, rule *k8s.CiliumNetworkPolicy) {
+			for n := 0; n < 10; n++ {
```

Comment:
Must be a const in `pkg/k8s/const.go` we should use a single value for all backoff loops so we can adjust globally.

Reply:
Fixed